firefox-ios/Client/Frontend/Browser/BrowserViewController/Extensions/BrowserViewController+WebViewDelegates.swift

```diff
@@ -33,6 +33,11 @@ extension BrowserViewController: WKUIDelegate {
             screenshotHelper.takeScreenshot(currentTab)
         }
 
+        if navigationAction.canOpenExternalApp, let url = navigationAction.request.url {
+            UIApplication.shared.open(url)
+            return nil
+        }
```

Comment:
Isn't this basically a similar fix to what was implemented for `whatsapp`? e.g.:

```
   if url.scheme == "whatsapp" && UIApplication.shared.canOpenURL(url) {
                UIApplication.shared.open(url, options: [:])
            }
```

If so I'm just curious why we're not handling this in the same way?

Reply:
It may look the same but is a little bit different... 
       
(few code for reference)
 ```
guard !navigationAction.isInternalUnprivileged,
              shouldRequestBeOpenedAsPopup(navigationAction.request)
        else {
            guard let url = navigationAction.request.url else { return nil }

            if url.scheme == "whatsapp" && UIApplication.shared.canOpenURL(url) {
                UIApplication.shared.open(url, options: [:])
            }

            return nil
        }
```
For other apps than "whatsapp" like "youtube" or "instagram", "shouldRequestBeOpenedAsPopup" method returns **true** because "url.scheme" is https not "youtube" or "instagram" as on "whatsapp" case. (based on the code, from above)

In this case, the logic, may be a little bit different.